mtz-auth-interceptor.html

```diff
@@ -0,0 +1,60 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="mtz-interceptor-behavior.html">
+
+<!--
+`mtz-auth-interceptor`
+An implementation of the mtz.InterceptorBehavior that injects auth headers based on the event.target containing the
+  [with-auth] attribute.
+-->
+<dom-module id="mtz-auth-interceptor">
+  <script>
+    Polymer({
+      is: 'mtz-auth-interceptor',
+      behaviors: [mtz.InterceptorBehavior],
+      properties: {
+        /* The events to intercept against */
+        events: {
+          type: Array,
+          value() {
+            return [
+              {
+                name: 'iron-ajax-presend',
+                path: 'detail.options.headers'
+              }
+            ];
+          }
+        },
+        /* The header name to attach to the ajax request */
+        key: {
+          type: String,
+          value: 'Authorization'
+        },
+        /* The value to associate with the provided header */
+        value: String
```

Comment:
I'm trying to match it to the presend-header as close as possible so it can easily be an extension in 2.0